llama_stack/distribution/resolver.py

```diff
@@ -299,6 +299,10 @@ async def instantiate_providers(
             api = Api(api_str)
             impls[api] = impl
 
+    # Add the list of implementations to the "providers" endpoint, which will later be used to
+    # retrieve the health status of each provider
+    impls[Api("providers")].impls = impls
```

Comment:
this trickery is not bueno. I think we should either:
- separate out these distro-specific implementations out of the resolver (or have a path for obtaining their implementation which is not tied to generic provider resolution)
- declare that `Api.providers` has a "dep" on all APIs 

I think the latter will become unnecessarily ugly.

Reply:
Why don't we just add implementations for both `Api.inspect` and `Api.providers` manually after all the other providers are resolved? Then you can just directly invoke these constructors and pass them whatever you want to pass them instead of having to go through `get_provider_impl`, etc.